Flesh to stone no longer makes you immune to bleeding, refactors bleedsuppress into a trait

CoffeeDragon16 opened the pull request:
## About The Pull Request

Being hit by the Flesh to Stone makes you bleed immune to prevent you from getting petrified while bleeding, and bleeding out while turned into stone. However, it doesn't make you vulnerable when you get unpetrified. This is a bug. This also makes bleedsuppress into a trait, as both is broken and should be a trait.

## Why It's Good For The Game

Bugs bad, traits good.

## Changelog
:cl:
fix: Flesh to stone no longer makes you bleed immune after you stop being a statue.
/:cl:

Commit:
Does the thing

Commit:
no i was right the first time

Commit:
Revert "no i was right the first time"

This reverts commit 72ea3004e08ce83c471314da6198a06ddedcc96d.

LemonInTheDark commented on code/game/objects/structures/petrified_statue.dm:
```diff
@@ -80,12 +80,16 @@
+	bleedsuppress = TRUE
 	S.copy_overlays(src)
 	var/newcolor = list(rgb(77,77,77), rgb(150,150,150), rgb(28,28,28), rgb(0,0,0))
 	S.add_atom_colour(newcolor, FIXED_COLOUR_PRIORITY)
+	addtimer(CALLBACK(src, .proc/unpetrify), statue_timer)
 	return TRUE
 
+/mob/living/carbon/human/proc/unpetrify()
+	bleedsuppress = FALSE
+
```
I take it statues don't have a "become not statue" proc you could hook into

CoffeeDragon16 replied:
Yup, the code is pretty old, it just deletes the statue at the end

CoffeeDragon16 replied:
Oh, turns out there was one that I just didn't see. Updated it to use that just now.

CoffeeDragon16 commented:
i was not right the first time

Commit:
refactors nobleed, actually works now

Commit:
timer for unpetrify

Commit:
Revert "organizes comments"

This reverts commit f211c56c13c924bbde857dab9ec750d8454ca30e.

Commit:
organizes comments

Commit:
merge conflict bang ding ow

coiax commented on code/__DEFINES/traits.dm:
```diff
@@ -100,24 +100,15 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
+#define TRAIT_UI_BLOCKED		"uiblocked" /// Inability to access UI hud elements. Turned into a trait from [MOBILITY_UI] to be able to track sources.
+#define TRAIT_PULL_BLOCKED		"pullblocked" /// Inability to pull things. Turned into a trait from [MOBILITY_PULL] to be able to track sources.
+#define TRAIT_RESTRAINED		"restrained" /// Abstract condition that prevents movement if being pulled and might be resisted against. Handcuffs and straight jackets, basically.
+#define TRAIT_PERFECT_ATTACKER "perfect_attacker" /// Doesn't miss attacks
```
100% personal opinion, but I think I prefer the comments before the variables, rather than on the same line, especially because some of these are REAL long.

Fikou replied:
this is literally breaking how autodoc works

CoffeeDragon16 replied:
ah, i undid it, i just thought it looked really bad like that

coiax commented on code/game/objects/structures/petrified_statue.dm:
```diff
@@ -80,12 +80,16 @@
 		return FALSE
 	var/obj/structure/statue/petrified/S = new(loc, src, statue_timer)
 	S.name = "statue of [name]"
-	bleedsuppress = 1
```
Where else is `bleedsuppress` used? You should probably replace the other instances with the trait.

CoffeeDragon16 replied:
Ah, thanks for reminding me. Updated it to totally replace it. It was used in highlander, blood trails, and handle_blood.

Commit:
finishes refactor

removes bleedsuppress entirely, gets rid of it in highlander, blood trails and handle_blood

Commit:
removes unpetrify() uses destroy()

Commit:
TWO remove_traits

moves it to two remove traits as it should be, assumemd you could've crammed em in

Rohesie commented:
`is_bleeding()` is not checking if the person has `TRAIT_NOBLEED`. Is that because you're guaranteeing that if they have the trait, the other checks will fail?

CoffeeDragon16 commented:
`is_bleeding` should always fail if you have `TRAIT_NOBLEED` because you can't bleed at all in the first place, yeah.

Rohesie commented:
No, it's better this way.
In fact, if we could eventually remove the need for an `is_bleeding()` proc altogether, that'd be better.

tgstation-server commented:
Your Fix/Feature pull request delta is now above zero (1). Feel free to make Feature/Tweak/Balance PRs.
